src/main/java/io/aiven/kafka/connect/opensearch/RecordConverter.java

```diff
@@ -119,17 +121,32 @@ private DocWriteRequest<?> createDocWriteRequest(final String indexOrDataStreamN
             return docIdStrategy.updateRequest(new DeleteRequest(indexOrDataStreamName), record);
         }
         final var payload = getPayload(record);
-        final var indexRequest = new IndexRequest().index(indexOrDataStreamName);
         if (config.dataStreamEnabled()) {
-            indexRequest
-                    .source(addTimestampToPayload(payload, record.timestamp()), XContentType.JSON)
-                    .opType(DocWriteRequest.OpType.CREATE);
+            return docIdStrategy.updateRequest(
+                    new IndexRequest()
+                            .index(indexOrDataStreamName)
+                            .opType(DocWriteRequest.OpType.CREATE)
+                            .source(addTimestampToPayload(payload, record.timestamp()), XContentType.JSON),
+                    record
+            );
+        } else if (config.indexWriteMethod() == IndexWriteMethod.UPSERT) {
+            return docIdStrategy.updateRequest(
```

Comment:
I think this is not going to work with `DocumentIDStrategy::NONE`: the update request needs `_id` and for `NONE`, it won't be assigned. It seems like we also need to add more test cases to cover `UpdateRequest` under all supported `DocumentIDStrategy`s

Reply:
We probably have to handle this configuration (`NONE` + `UPSERT`) as non valid.

Reply:
sure makes sense. will add a check for this. for update we need to use a record key as a doc id